.github/workflows/codeql-buildscript.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
```

Comment:
I would rather not add another build script, as that just makes it harder whenever we change something in the build procedure. Is there a way we can use the Docker image or somehow use the existing workflow for building the project?

Reply:
Updated to include build instructions in the existing `codeql.yml` workflow.

Reply:
I think you misunderstood. I didn't mean to just move the code from the script to the workflow file. I'm not entirely sure what the point of building the firmware in the codeql workflow is. Is it to have the workflow fail if the firmware doesn't build? If so, isn't it possible to have the codeql workflow only run if the build-firmware workflow succeeded?

Reply:
I see. Source code needs to be built: https://docs.github.com/en/code-security/code-scanning/troubleshooting-code-scanning/no-source-code-seen-during-build. If there's another workflow that builds this repo, unfortunately this workflow won't be able to analyze the build from that one. This is a limitation, but we don't believe it really hinders use.

Reply:
I see. Thanks for the link! Can you also use Docker to build it, or does that not work with CodeQL? I would prefer that to help with maintainability.

Reply:
We have the option of adding the following lines to the wf:

```
# Specify the container in which actions will run
    container: 
      image: mcr.microsoft.com/cstsectools/codeql-container
      options: --user root
```

(from https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/running-codeql-code-scanning-in-a-container)